mark_outliers_in_json.py

```diff
@@ -35,27 +35,52 @@ def main(in_files, window_size):
         krun_data[filename] = read_krun_results_file(filename)
         krun_data[filename]['window_size'] = window_size
     for filename in krun_data:
-        outliers = dict()
-        for benchmark in krun_data[filename]['data']:
-            outliers[benchmark] = list()
-            for p_exec in krun_data[filename]['data'][benchmark]:
-                outliers[benchmark].append(find_outliers(p_exec, window_size))
-        krun_data[filename]['outliers'] = outliers
+        all_outliers = dict()
+        unique_outliers = dict()
+        common_outliers = dict()
+        for bench in krun_data[filename]['data']:
+            all_outliers[bench] = list()
+            for p_exec in krun_data[filename]['data'][bench]:
+                all_outliers[bench].append(get_all_outliers(p_exec, window_size))
+            common, unique = get_outliers(all_outliers[bench],
+                                          window_size,
+                                          threshold)
+            common_outliers[bench] = common
+            unique_outliers[bench] = unique
+        krun_data[filename]['all_outliers'] = all_outliers
+        krun_data[filename]['common_outliers'] = common_outliers
+        krun_data[filename]['unique_outliers'] = unique_outliers
         new_filename = create_output_filename(filename, window_size)
         print('Writing out: %s' % new_filename)
         write_krun_results_file(krun_data[filename], new_filename)
 
 
-def find_outliers(data, window_size):
-    outliers = list()
+def get_outliers(all_outliers, window_size, threshold=1):
+    common, unique = list(), list()
+    for index, outliers in enumerate(all_outliers):
+        common_exec = list()
+        unique_exec = list()
+        for outlier in outliers:
+            other_execs = all_outliers[:index] + all_outliers[(index + 1):]
+            if sum(map(lambda lst: outlier in lst, other_execs)) > threshold:
```

Comment:
I have two issues with this line: a) I think it should be ``... >= threshold`` otherwise 2 other executions are needed, not 1 for threshold = 1.

b) as usual I find this kind of code pretty hard to wrap my head around. Why not simply
```
sum = 0
for execution in other_execs:
    sum += outlier in execution
if sum > threshold:
    ...
```